build/config.gni

```diff
@@ -78,45 +79,79 @@ if (is_win) {
   _channel = ""
-  brave_app_guid = "{AFE6A462-C574-4B8A-AF43-4CC60DF4563B}"
-  if (brave_channel == "nightly") {
-    # Temporary support PR's non-Release builds on nightly channel.
-    # CI expect the installer files (*Setup.exe) in the output dir are
-    # correspond passed brave_channel. Therefore to make Static+nightly work
-    # we should set _channel to "Nightly".
-    # TODO(atuchin): revert after PRs start to use the development channel.
-    _channel = "Nightly"
-    brave_app_guid = "{C6CB981E-DB30-4876-8639-109F8933582C}"
-  } else if (is_official_build) {
-    if (brave_channel == "beta") {
-      _channel = "Beta"
-      brave_app_guid = "{103BD053-949B-43A8-9120-2E424887DE11}"
-    } else if (brave_channel == "dev") {
-      _channel = "Dev"
-      brave_app_guid = "{CB2150F2-595F-4633-891A-E39720CE0531}"
+
+  # Brave Origin uses separate app GUIDs from Brave Browser to allow
+  # side-by-side installation and independent update infrastructure.
+  if (is_brave_origin_branded) {
+    brave_app_guid = "{F1EF32DE-F987-4289-81D2-6C4780027F9B}"
+    if (brave_channel == "nightly") {
+      _channel = "Nightly"
+      brave_app_guid = "{50474E96-9CD2-4BC8-B0A7-0D4B6EF2E709}"
+    } else if (is_official_build) {
+      if (brave_channel == "beta") {
+        _channel = "Beta"
+        brave_app_guid = "{56DA94FD-D872-416B-BFC4-1D7011DA7473}"
+      } else if (brave_channel == "dev") {
+        _channel = "Dev"
+        brave_app_guid = "{716D6A4A-D071-47A8-AC64-DBDE3EE3797B}"
+      } else {
+        assert(brave_channel == "", "Unknown channel name")
+      }
     } else {
-      assert(brave_channel == "", "Unknown channel name")
+      _channel = "Development"
     }
   } else {
-    _channel = "Development"
+    brave_app_guid = "{AFE6A462-C574-4B8A-AF43-4CC60DF4563B}"
+    if (brave_channel == "nightly") {
+      # Temporary support PR's non-Release builds on nightly channel.
+      # CI expect the installer files (*Setup.exe) in the output dir are
+      # correspond passed brave_channel. Therefore to make Static+nightly work
+      # we should set _channel to "Nightly".
+      # TODO(atuchin): revert after PRs start to use the development channel.
+      _channel = "Nightly"
+      brave_app_guid = "{C6CB981E-DB30-4876-8639-109F8933582C}"
+    } else if (is_official_build) {
+      if (brave_channel == "beta") {
+        _channel = "Beta"
+        brave_app_guid = "{103BD053-949B-43A8-9120-2E424887DE11}"
+      } else if (brave_channel == "dev") {
+        _channel = "Dev"
+        brave_app_guid = "{CB2150F2-595F-4633-891A-E39720CE0531}"
+      } else {
+        assert(brave_channel == "", "Unknown channel name")
+      }
+    } else {
+      _channel = "Development"
+    }
   }
+
   _arch = ""
   if (target_cpu == "x86") {
     _arch = "32"
   }
+
+  # Use different installer executable names for Brave Origin
+  if (is_brave_origin_branded) {
+    _installer_brand = "BraveOrigin"
```

Comment:
To the extent possible, I think it would be nicer to put Origin's artifacts at the same output paths as Brave's. The reason is the following: If we put them at different paths, then both brave-core and the devops code that takes the artifacts need to have the logic "is this Origin? If yes, use path X, otherwise use path Y". It's duplicate logic that's tedious to work with and keep track of. I understand we want "Origin" to appear in user-facing paths and URLs. But (imo) that should be done at the outermost layer of the onion.

The `$_channel` and `$brave_underline_full_version` usages in the existing code are examples of how it should not be done (imo).

Reply:
I'd like for app names and install names to be different.